programs/witness_node/main.cpp

```diff
@@ -127,6 +113,10 @@ int main(int argc, char** argv) {
          return 0;
       }
 
+      cfg_options.add_options()
+              ("plugins", bpo::value<std::string>()->default_value(options.at("plugins").as<std::string>()),
+               "Space-separated list of plugins to activate");
```

Comment:
What if we move this to be before `auto witness_plug = node->register_plugin<witness_plugin::witness_plugin>();`? I guess `plugins` will be be moved up a bit in `config.ini`.

Reply:
that will put the plugin option at the top of the config file.

Reply:
i think i have it, a bit dirty now so cleaning up, will commit in a bit.

Reply:
Got it. Actually if we want `plugins` to be before plugin options and after other application options, we need to find a way to do it after https://github.com/bitshares/bitshares-core/blob/f1e448059ae5861f56ce81f23876770fc134dbd7/libraries/app/application.cpp#L955-L978 but before https://github.com/bitshares/bitshares-core/blob/f1e448059ae5861f56ce81f23876770fc134dbd7/libraries/app/application.cpp#L989.

Reply:
it can be done in the binary if we first add the non plugin options, then load the plugins and add the options again for them as:

https://github.com/bitshares/bitshares-core/pull/1640/commits/d060bbd4b2133e5c97c40586a9c965572d643377

This way config will look as:

```
...
# Whether allow API clients to subscribe to universal object creation and removal events
# enable-subscribe-to-all = 

# Whether to enable tracking of votes of standby witnesses and committee members. Set it to true to provide accurate data to API clients, set to false for slightly better performance.
# enable-standby-votes-tracking = 

# Space-separated list of plugins to activate
plugins = witness account_history market_history grouped_orders

# Enable block production, even if the chain is stale.
enable-stale-production = false

# Percent of witnesses (0-99) that must be participating in order to produce blocks
required-participation = false
...
```

That is with `plugins` on top of the first `witness` plugin option.

Reply:
Looks good. I don't know whether there are side effects.